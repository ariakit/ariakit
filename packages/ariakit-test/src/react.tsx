import type { ReactElement } from "react";
import { StrictMode } from "react";
import * as ReactTestingLibrary from "@testing-library/react";
import { flushMicrotasks, nextFrame, wrapAsync } from "./__utils.js";

export * from "./index.js";

export interface RenderOptions
  extends Omit<ReactTestingLibrary.RenderOptions, "queries"> {
  strictMode?: boolean;
}

export async function render(ui: ReactElement, options?: RenderOptions) {
  const wrapper = (props: { children: ReactElement }) => {
    const Wrapper = options?.wrapper;
    const element = Wrapper ? <Wrapper {...props} /> : props.children;
    if (!options?.strictMode) return element;
    return <StrictMode>{element}</StrictMode>;
  };

  return wrapAsync(async () => {
    const { unmount } = ReactTestingLibrary.render(ui, { ...options, wrapper });
    await flushMicrotasks();
    await nextFrame();
    await flushMicrotasks();
    return unmount;
  });
}
