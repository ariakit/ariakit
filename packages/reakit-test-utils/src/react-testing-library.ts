export {
  act,
  fireEvent,
  render,
  wait,
  waitForDomChange,
  waitForElement,
  waitForElementToBeRemoved
} from "@testing-library/react";
