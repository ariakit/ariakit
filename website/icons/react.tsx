import type { SVGAttributes } from "react";

export function React(props: SVGAttributes<SVGSVGElement>) {
  return (
    <svg viewBox="-11.5 -10.23174 23 20.46348" {...props}>
      <circle
        className="fill-blue-600 dark:fill-[#61dafb]"
        cx="0"
        cy="0"
        r="2.05"
      />
      <g
        className="stroke-blue-600 dark:stroke-[#61dafb]"
        strokeWidth="1"
        fill="none"
      >
        <ellipse rx="11" ry="4.2" />
        <ellipse rx="11" ry="4.2" transform="rotate(60)" />
        <ellipse rx="11" ry="4.2" transform="rotate(120)" />
      </g>
    </svg>
  );
}
