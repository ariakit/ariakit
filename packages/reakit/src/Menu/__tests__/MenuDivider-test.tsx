import * as React from "react";
import { render } from "react-testing-library";
import MenuDivier from "../MenuDivider";

test("html attrs", () => {
  const { getByTestId } = render(
    <MenuDivier id="test" aria-label="test" data-testid="test" />
  );
  expect(getByTestId("test")).toHaveAttribute("id", "test");
  expect(getByTestId("test")).toHaveAttribute("aria-label", "test");
});

test("styled", () => {
  const { container } = render(<MenuDivier />);
  expect(container.firstChild).toMatchSnapshot();
});
