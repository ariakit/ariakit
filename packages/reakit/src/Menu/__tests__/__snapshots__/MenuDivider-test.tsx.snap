// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled 1`] = `
.c1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
  background-color: unset;
  color: inherit;
}

.c0 {
  color: transparent;
  font-size: 0;
  height: 2px;
  width: 100%;
}

[aria-orientation="horizontal"] > .c0 {
  height: unset;
  width: 2px;
}

<div
  class="c0 c1"
/>
`;
